pkg/processor/procbuiltin/unwraprecord.go

```diff
@@ -0,0 +1,174 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procbuiltin
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/multierror"
+	"github.com/conduitio/conduit/pkg/processor"
+	"github.com/conduitio/conduit/pkg/record"
+)
+
+const (
+	unwrapRecordPayloadProcType = "unwraprecord"
```

Comment:
I think `convert` is more about converting the whole record to a new format (which is what we do in the middleware), but here, we ARE taking out something and removing layers, we're taking out the record from the `rec.Payload.After `and returning that as the new record, the input record is openCDC, so is the output record. so for me `unwrap` makes sense

Reply:
I think both of you make good points and both names would work. I'm leaning towards `unwrap` though, because it will be easier for us to distinguish them when we talk about "converting" and "unwrapping".